connector/src/main/scala/com/datastax/bdp/spark/ContinuousPagingScanner.scala

```diff
@@ -22,16 +24,13 @@ import com.datastax.spark.connector.util._
 case class ContinuousPagingScanner(
   readConf: ReadConf,
   connConf: CassandraConnectorConf,
-  columnNames: IndexedSeq[String]) extends Scanner with Logging {
+  columnNames: IndexedSeq[String],
+  cqlSession: CqlSession) extends Scanner with Logging {
 
   val TARGET_PAGE_SIZE_IN_BYTES: Int = 5000 * 50 // 5000 rows * 50 bytes per row
   val MIN_PAGES_PER_SECOND = 1000
 
-  //TODO This must be moved to session initilization? We can no longer pass options at execution time without deriving a new profile
-  // I think the right thing to do, to support old configurations as well as new is to create a new profile based on options as
-  // Set using derivied profiles, but this probably can't happen here
-  /**
-  private val cpOptions =  readConf.throughputMiBPS match {
+  private lazy val cpProfile = readConf.throughputMiBPS match {
     case Some(throughput) =>
       val bytesPerSecond = (throughput * 1024 * 1024).toInt
```

Comment:
Question: shoud we use `toInt` here? I see the other classes uses `toLong`, e.g.
```
(throughput * 1024 * 1024).toLong
```
Also, do we have a max cap here?

Reply:
Int.MaxValue is the cap here. I'll make sure we have a meaningful exception here